src/yajl_buf.c

```diff
@@ -68,15 +80,19 @@ void yajl_buf_free(yajl_buf buf)
     YA_FREE(buf->alloc, buf);
 }
 
-void yajl_buf_append(yajl_buf buf, const void * data, size_t len)
+int yajl_buf_append(yajl_buf buf, const void * data, size_t len)
 {
-    yajl_buf_ensure_available(buf, len);
+    if (-1 == yajl_buf_ensure_available(buf, len))
```

Comment:
it seems like all of these could be `if ( fn() < 0 )` which is slightly more idiomatic C